src/components/MainPage/Listing.vue

```diff
@@ -0,0 +1,68 @@
+<template>
+    <div class="mainListing">
+        <div class="switch">
+            <img v-if="view === 'Grid'" class="switchClick" src="../../assets/icons/grid.svg" />
+            <img v-if="view === 'List'"  v-on:click="setView('Grid')" class="switchClick"  src="../../assets/icons/gridx.svg" />
+            <img v-if="view === 'List'" class="switchClick" src="../../assets/icons/list.svg" />
+            <img v-if="view === 'Grid'" v-on:click="setView('List')" class="switchClick"  src="../../assets/icons/listx.svg" />        
+        </div>
+        <Grid v-if="view === 'Grid'"/>
+        <List v-if="view === 'List'"/>
+    </div>    
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import { mapState } from 'vuex'
+import Grid from './ListingComponents/Grid.vue'
+import List from './ListingComponents/List.vue'
+
+@Component({
+    components: {
+        Grid,
+        List
+    },
+    // mounted() {
+    //     console.log('wasssup')
+    // },
+    computed: {
+        ...mapState({
+            // eslint-disable-next-line 
+            view: (state: any) => state.listings.view
```

Comment:
Is it possible for you to create an interface for the state, this way we wouldn't need to disable eslint,
https://codeburst.io/vuex-and-typescript-3427ba78cfa8
this article demonstrates a nice way of doing so.